cpu/fe310/periph/spi.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2019 Tristan Bruns
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_fe310
+ * @ingroup     drivers_periph_spi
+ *
+ * @{
+ *
+ * @file        spi.c
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Tristan Bruns
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "assert.h"
+#include "periph/spi.h"
+#include "vendor/platform.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+/**
+ * @brief   Allocation device locks
+ */
+static mutex_t lock;
+
+void spi_init(spi_t bus)
+{
+    /* make sure given bus device is valid */
+    assert(bus == 0);
+
+    /* initialize the buses lock */
+    mutex_init(&lock);
+
+    /* trigger pin initialization */
+    spi_init_pins(bus);
+
+    /* disable hardware chip select
+       (hardware chip select only supports one-byte transfers...) */
+    SPI1_REG(SPI_REG_CSMODE) = SPI_CSMODE_OFF;
+}
+
+void spi_init_pins(spi_t bus)
+{
+    (void) bus;
+    assert(bus == 0);
+
+    const gpio_t spi1_pins =
+        (1 << IOF_SPI1_MOSI) |
+        (1 << IOF_SPI1_MISO) |
+        (1 << IOF_SPI1_SCK);
+
+    /* Enable I/O Function 0 */
+    GPIO_REG(GPIO_IOF_EN)  |=  spi1_pins;
+    GPIO_REG(GPIO_IOF_SEL) &= ~spi1_pins;
+}
+
+int spi_init_cs(spi_t bus, spi_cs_t cs)
+{
+    if (bus != 0) {
+        return SPI_NODEV;
+    }
+
+    /* setting the CS high before configuring it as an
+       output should be fine on FE310. */
+    gpio_set(cs);
+
+    if (gpio_init(cs, GPIO_OUT)) {
+        return SPI_NOCS;
+    }
+
+    return SPI_OK;
+}
+
+int spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
+{
+    (void) bus;
```

Comment:
I think it would make life easier to also add `assert()`s for `bus` and `cs` here, so that developers are supported in finding their bugs in using the SPI interface.